mixer/pkg/runtime/config/ephemeral.go

```diff
@@ -353,7 +355,8 @@ func (e *Ephemeral) processDynamicHandlerConfigs(ctx context.Context, adapters m
 			// validate if the param is valid
 			bytes, err := validateEncodeBytes(hdl.Params, adapter.ConfigDescSet, getParamsMsgFullName(adapter.PackageName))
 			if err != nil {
-				appendErr(ctx, errs, fmt.Sprintf("handler='%s'.params", handlerName), monitoring.HandlerValidationErrors, err.Error())
+				validationErrs++
+				appendErr(errs, fmt.Sprintf("handler='%s'.params", handlerName), err.Error())
```

Comment:
I'm not a fan of this pattern. It would be better if appendErr would own the error counters internally as part of error structure. We could add a parameter here for the type of the error, and then dump the counters at the end. This is more of a stylistic issue, so feel free to disregard.